src/withFinalForm/withFinalForm.tsx

```diff
@@ -9,88 +8,70 @@ import { UseFinalFormFieldValue } from '../useFinalFormField/useFinalFormField.i
+        'value' | keyof FieldTransformationProps<any, any, any>
       >
 
-    type FieldComponentProps = Omit<
-      BaseProps,
-      keyof FieldTransformationProps<InputValue, BaseProps>
-    > &
-      FieldTransformationProps<InputValue, BaseProps>
-
-    const hookConfig = {
-      changeOnBlur: inputConfig.changeOnBlur,
-    }
-
-    const useFieldProps = ({
-      format: rawFormat,
-      parse: rawParse,
-      validate: rawValidate,
-      errorBehavior: rawErrorBehavior,
-      ...props
-    }: FieldComponentProps) => {
+    type ComponentProps = BaseProps &
+      FieldTransformationProps<InputValue, FieldValue, BaseProps>
+
+    return React.forwardRef<Element, ComponentProps>((props, ref) => {
+      const fieldPropsRef = React.useRef<UseFinalFormFieldValue<FieldValue>>()
+      const propsRef = React.useRef(props)
       const t = useTranslate()
-      const propsRef = React.useRef<BaseProps>(props as BaseProps)
-      const callbackRef =
-        React.useRef<FieldTransformationProps<InputValue, BaseProps>>()
 
-      propsRef.current = props as BaseProps
-      callbackRef.current = {
-        format: rawFormat,
-        parse: rawParse,
-        validate: rawValidate,
-      } as FieldTransformationProps<InputValue, BaseProps>
+      propsRef.current = props
 
       const format = React.useCallback((value) => {
-        const fieldFormattedValue = isFunction(inputConfig.format)
-          ? inputConfig.format(value, propsRef.current)
-          : defaultFormat(value)
-
-        return isFunction(callbackRef.current?.format)
-          ? callbackRef.current!.format(fieldFormattedValue, propsRef.current)
-          : fieldFormattedValue
+        let result = value
+
+        // The default format comes directly from [React Final Form](https://github.com/final-form/react-final-form/blob/464f1c7855e93899630df0ad897c322995601849/src/useField.js#L24).
+        if (isFunction(options.format)) {
+          result = options.format(value, propsRef.current)
+        } else if (value === undefined) {
+          result = ''
+        }
+
+        return isFunction(propsRef.current?.format)
+          ? propsRef.current.format(result, propsRef.current)
+          : result
       }, [])
 
       const parse = React.useCallback((value) => {
-        const fieldParsedValue = isFunction(inputConfig.parse)
-          ? inputConfig.parse(value, propsRef.current)
-          : defaultParse(value)
-
-        return isFunction(callbackRef.current?.parse)
-          ? callbackRef.current!.parse(fieldParsedValue, propsRef.current)
-          : fieldParsedValue
+        let result = value
+
+        // Override the default parse method in an attempt to mitigate [this issue](https://github.com/final-form/react-final-form/issues/130).
+        if (isFunction(options.parse)) {
+          result = options.parse(value, propsRef.current)
+        } else if (value === '') {
+          result = fieldPropsRef.current?.meta.initial && ''
```

Comment:
```suggestion
          // Set initial value instead of `''` for `null` or `undefined` initial values
          result = fieldPropsRef.current?.meta.initial && ''
```